### ISSUE TITLE
No R completions on VSCode extension

### ISSUE THREAD

**Issue by PauloCampana** [quarto-dev/quarto#437]
Related:
https://github.com/quarto-dev/quarto/issues/329
https://github.com/quarto-dev/quarto/issues/190
https://github.com/quarto-dev/quarto-cli/discussions/4269

Can't get R code completions/suggestions to work inside an {r} chunk of a .qmd file.

It looks like the quarto VSCode extension creates a `.intellisense.r` file (sometimes a `.vdoc.r`?), grabs completions from there through the REditorSupport extension then serve them in the current active quarto editor, creating then trashing the file on every request.

From the discussion in quarto-cli linked above, it seems like this file should be created in a nearby directory, but mine is being created at `/tmp/tmp-*-*/r/` directory, and with no write permission to that root folder, the temporary file can't be deleted.

This means that I get 1 code completion right and after that no more until it decides to create the file in another tmp directory which happens when VSCode is restarted. Here is the debug logs from Output -> Window:

```
2024-05-10 20:09:25.502 [debug] _performFileEdits [{"oldResource":{"$mid":1,"fsPath":"/tmp/tmp-26098-KsrUF66R9pC5/r/.intellisense.r","external":"file:///tmp/tmp-26098-KsrUF66R9pC5/r/.intellisense.r","path":"/tmp/tmp-26098-KsrUF66R9pC5/r/.intellisense.r","scheme":"file"},"options":{}}]
2024-05-10 20:09:25.517 [error] Unknown (FileSystemError): Failed to move '.intellisense.r' to the trash (Failed to move item to trash)
    at $.create (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:56:3584)
    at P (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:56:3698)
    at b.y (/usr/share/code/resources/app/out/vs/code/electron-main/main.js:61:40216)
2024-05-10 20:09:25.517 [warning] IGNORING workspace edit: Unknown (FileSystemError): Failed to move '.intellisense.r' to the trash (Failed to move item to trash)
```

Quarto extension: v1.111.0
R extension: v2.8.3
languageserver package: v0.3.16
VSCode: v1.89.1-1715060508
Linux Mint: v21.3

**Comment by hudtorrent** [comment]
PauloCampana, your observation seems correct.

I have a machine with the problem you described and another machine where things work correctly.

On the machine where things work the file ".intellisense.r" is located at "/.local/share/Trash/files".

Can we change where the intellisense file is created and fix the problem?

**Comment by jaschaegli** [comment]
TLDR: 
I found two functioning workarounds: 

1: Install trash-cli and call vscode after setting `ELECTRON_TRASH=trash-cli`. 

2: Set `"files.enableTrash": false`, if you're okay with _all_ deletions by vscode being permanent, i.e. bypassing the trash entirely. . 

---
Long version: 

I'm running into the same issue with vscodium 1.93.1, quarto 1.5.57, quarto-vscode v1.115.0, and languageserver 0.3.16, running on Arch Linux (6.10.10-arch1-1)

```
2024-09-26 15:15:49.939 [error] Unknown (FileSystemError): Failed to move '.intellisense.r' to the trash (Failed to move item to trash)
    at D.create (/opt/vscodium-bin/resources/app/out/vs/code/electron-main/main.js:63:4609)
    at w (/opt/vscodium-bin/resources/app/out/vs/code/electron-main/main.js:63:4749)
    at u.delete (/opt/vscodium-bin/resources/app/out/vs/code/electron-main/main.js:69:45991)
2024-09-26 15:15:49.940 [warning] IGNORING workspace edit: Unknown (FileSystemError): Failed to move '.intellisense.r' to the trash (Failed to move item to trash)
```

Manually moving the intellisense.r file from the tmp subdirectory to ~/.local/share/Trash works fine from a terminal. What _doesn't_ work is attempting to trash it using `gio`, which Electron [uses for trashing by default](https://github.com/electron/electron/blob/main/docs/api/environment-variables.md#electron_trash-linux). Doing so fails with `Trashing on system internal mounts is not supported`. As far as I can tell, the problem is thus not with any actual file permissions, but with the fact that (1) gio simply refuses to trash anything from /tmp/, and (2) quarto doesn't fall back to simply deleting if a trashing operation has failed. 

Two fixes that work for me are to circumvent the trashing issue entirely by disabling trashing altogether (`"files.enableTrash": false`), or to install a trashing backend that does support trashing things from /tmp/ (trash-cli) and instructing vscode to use this instead of gio by setting `ELECTRON_TRASH=trash-cli`.

**Comment by hudtorrent** [comment]
Thanks a lot, @jaschaegli!

I tested option 2 and it worked!

**Comment by juliasilge** [comment]
This isn't specific to the vscode-R extension completions, as we also have someone reporting this problem in https://github.com/posit-dev/positron/issues/7471 in Positron, using remote SSH.
